### ISSUE TITLE
Calico Typha pods Landing on fargate

### ISSUE THREAD

**Issue by abdallah-refaay** [tigera/operator#2310]
I am deploying calico on EKS with the latest tigera operator helm chart 3.24.5 operator version 1.28.5,
I added nodeSelector for the operator to land on specific nodes.

My values:
installation.kubernetesProvider="EKS"
nodeSelector."karpenter\.sh/initialized"=true

## Expected Behavior
Typha pods landing on karpenter node

## Current Behavior
all calico deployment pods landing on karpenter nodes except for typha, hundreds of typha pods are provisioned on fargate and it does not stop provisioning unless I delete all the resources in calico-system namespace manually.
![image](https://user-images.githubusercontent.com/59100389/203790217-7eaae6d6-b6fc-478c-9192-f9f641411291.png)

## Possible Solution
- typha should have affinity just like the operator not to be scheduled on fargate if the provider value is "EKS", just as done in this PR: https://github.com/tigera/operator/pull/1725
- adding nodeSelector or node affinity for typha deployment.

## Your Environment

* Operating System and version: EKS 1.23

**Comment by wmagon** [comment]
Hi,
have you tried setting Typha Affinity through Tigera Installation Custom Resource?
https://docs.tigera.io/calico/latest/reference/installation/api#operator.tigera.io/v1.TyphaDeploymentPodSpec
It seems possible to parameterize Deployment spec.

**Comment by joshuaganger** [comment]
Just ran into this issue on EKS 1.25 after installing via the helm repo as described here: https://docs.tigera.io/calico/latest/getting-started/kubernetes/helm#install-calico  It seems everything is working as expected on the EC2 nodes, but the Typha pods can't start on Fargate with an UnsupportedPodSpec error, and message `Pod not supported: SchedulerName is not fargate-scheduler, fields not supported: HostNetwork, port contains HostIP or HostPort`

Could you provide a little more detail around how to set the "Typha Affinity"?  It seems like this same issue was resolved for calico-node daemonset, shouldn't this also be default behavior to be the same for the calico-typha deployment?

**Comment by wmagon** [comment]
In case of calico-node that was perhaps simpler, as Fargate cannot work with DaemonSets and it looks EKS admissions are not replacing default-scheduler with fargate-scheduler. 
In case of calico-typha my previous suggestion is not much helpful for Fargate issues, even though Affinity can be set, it's ignored by the scheduler, just as Tolerantions are. 
More detailed descriptions can be found here:
https://github.com/aws/containers-roadmap/issues/998
https://github.com/aws/containers-roadmap/issues/1324

The simplest solution is probably just to not create Fargate profiles for calico namespaces. Calico can't work on Fargate anyway, so avoiding profiles for namespaces (or matching labels) like `tigera-operator`, `calico-system`, `calico-apiserver` should make sure it won't be deployed there.

**Comment by abdallah-refaay** [comment]
> namespaces

yes you're right calico namespaces don't even have a fargate profile even though typha pods are able to get scheduled on fargate.
